pkg/pool-weighted/test/foundry/LBPool.t.sol

```diff
@@ -0,0 +1,477 @@
+        assertEq(
+            balances[tokenIndexOut],
+            tokenAmounts[tokenIndexOut] - amountCalculated,
+            "Pool: Wrong tokenOut balance"
+        );
+    }
+
+    function testOnlyOwnerCanBeLP() public {
+        uint256[] memory amounts = [TOKEN_AMOUNT, TOKEN_AMOUNT].toMemoryArray();
+
+        vm.startPrank(bob);
+        router.addLiquidityUnbalanced(address(pool), amounts, 0, false, "");
+        vm.stopPrank();
+
+        vm.startPrank(alice);
+        vm.expectRevert(abi.encodeWithSelector(IVaultErrors.BeforeAddLiquidityHookFailed.selector));
+        router.addLiquidityUnbalanced(address(pool), amounts, 0, false, "");
+        vm.stopPrank();
+    }
+
+    function testSwapRestrictions() public {
+        // Ensure swaps are initially enabled
+        assertTrue(LBPool(address(pool)).getSwapEnabled(), "Swaps should be enabled initially");
+
+        // Test swap when enabled
+        vm.prank(alice);
+        router.swapSingleTokenExactIn(
+            address(pool),
+            IERC20(dai),
+            IERC20(usdc),
+            TOKEN_AMOUNT / 10,
+            0,
+            block.timestamp + 1 hours,
+            false,
+            ""
+        );
+
+        // Disable swaps
+        vm.prank(bob);
+        LBPool(address(pool)).setSwapEnabled(false);
+
+        // Verify swaps are disabled
+        assertFalse(LBPool(address(pool)).getSwapEnabled(), "Swaps should be disabled");
+
+        // Test swap when disabled
+        vm.prank(alice);
+        vm.expectRevert(abi.encodeWithSelector(IVaultErrors.BeforeSwapHookFailed.selector));
+        router.swapSingleTokenExactIn(
+            address(pool),
+            IERC20(dai),
+            IERC20(usdc),
+            TOKEN_AMOUNT / 10,
+            0,
+            block.timestamp + 1 hours,
+            false,
+            ""
+        );
+
+        // Re-enable swaps
+        vm.prank(bob);
+        LBPool(address(pool)).setSwapEnabled(true);
+
+        // Verify swaps are re-enabled
+        assertTrue(LBPool(address(pool)).getSwapEnabled(), "Swaps should be re-enabled");
+
+        // Test swap after re-enabling
+        vm.prank(alice);
+        router.swapSingleTokenExactIn(
+            address(pool),
+            IERC20(dai),
+            IERC20(usdc),
+            TOKEN_AMOUNT / 10,
+            0,
+            block.timestamp + 1 hours,
+            false,
+            ""
+        );
+    }
+
+    function testSetSwapFeeTooLow() public override {
+        vm.prank(bob);
+        vm.expectRevert(IVaultErrors.SwapFeePercentageTooLow.selector);
+        vault.setStaticSwapFeePercentage(pool, poolMinSwapFeePercentage - 1);
+    }
+
+    function testSetSwapFeeTooHigh() public override {
+        vm.prank(bob);
+        vm.expectRevert(abi.encodeWithSelector(IVaultErrors.SwapFeePercentageTooHigh.selector));
+        vault.setStaticSwapFeePercentage(pool, poolMaxSwapFeePercentage + 1);
+    }
```

Comment:
I had originally defined these tests here as `override`s, but ultimately decided it was cleaner to have `BasePoolTest` handle them with a more intelligent way to handle the swap fee manager. Is it preferable to have this in this file? Feels like needlessly duplicating code to me